tests/Feature_v2/Album/SharingTest.php

```diff
@@ -74,4 +77,127 @@ public function testUserGet(): void
+		]);
+		$this->assertNoContent($response);
+		self::assertEquals(1, AccessPermission::where(APC::BASE_ALBUM_ID, '=', $this->subAlbum1->id)->count());
+		$perm = AccessPermission::where(APC::BASE_ALBUM_ID, '=', $this->subAlbum1->id)->first();
+
+		// Update the permission with false
+		$response = $this->actingAs($this->userMayUpload1)->patchJson('Sharing', [
+			'perm_id' => $perm->id,
+			'grants_edit' => false,
+			'grants_delete' => false,
+			'grants_download' => false,
+			'grants_full_photo_access' => false,
+			'grants_upload' => false,
+		]);
+		$this->assertOk($response);
+
+		// Verify the permission
+		$perm = AccessPermission::where(APC::BASE_ALBUM_ID, '=', $this->subAlbum1->id)->first();
+		self::assertFalse($perm->grants_edit);
+		self::assertFalse($perm->grants_delete);
+		self::assertFalse($perm->grants_download);
+		self::assertFalse($perm->grants_full_photo_access);
+		self::assertFalse($perm->grants_upload);
+
+		// Apply update again
+		$response = $this->actingAs($this->userMayUpload1)->putJson('Sharing', [
+			'album_id' => $this->album1->id,
+			'shall_override' => false,
+		]);
+		$this->assertNoContent($response);
+		// Verify the count is still 1.
+		self::assertEquals(1, AccessPermission::where(APC::BASE_ALBUM_ID, '=', $this->subAlbum1->id)->count());
+
+		// Verify the permission
+		$perm = AccessPermission::where(APC::BASE_ALBUM_ID, '=', $this->subAlbum1->id)->first();
+		self::assertTrue($perm->grants_edit);
+		self::assertTrue($perm->grants_delete);
+		self::assertTrue($perm->grants_download);
+		self::assertTrue($perm->grants_full_photo_access);
+		self::assertTrue($perm->grants_upload);
+	}
+
+	public function testOverride(): void
+	{
+		// Set up the permission in subSlbum
+		$response = $this->actingAs($this->userMayUpload1)->postJson('Sharing', [
+			'user_ids' => [$this->userLocked->id],
+			'album_ids' => [$this->subAlbum1->id],
+			'grants_edit' => true,
+			'grants_delete' => true,
+			'grants_download' => true,
+			'grants_full_photo_access' => true,
+			'grants_upload' => true,
+		]);
+		$this->assertOk($response);
+		self::assertEquals(1, AccessPermission::where(APC::BASE_ALBUM_ID, '=', $this->subAlbum1->id)->count());
+
+		$response = $this->actingAs($this->userMayUpload1)->postJson('Sharing', [
+			'user_ids' => [$this->userNoUpload->id],
+			'album_ids' => [$this->album1->id],
+			'grants_edit' => true,
+			'grants_delete' => true,
+			'grants_download' => true,
+			'grants_full_photo_access' => true,
+			'grants_upload' => true,
+		]);
+		$this->assertOk($response);
+		self::assertEquals(2, AccessPermission::where(APC::BASE_ALBUM_ID, '=', $this->album1->id)->count());
+
+		// Update sub album permission.
+		$response = $this->actingAs($this->userMayUpload1)->putJson('Sharing', [
+			'album_id' => $this->album1->id,
+			'shall_override' => true,
+		]);
+		$this->assertNoContent($response);
+		// self::assertEquals(2, AccessPermission::where(APC::BASE_ALBUM_ID, '=', $this->subAlbum1->id)->count());
```

Comment:
Commented code. Is this required or can it be cleaned up and removed?

Reply:
Good spot! 
Fixed [9d66a20](https://github.com/LycheeOrg/Lychee/pull/2950/commits/9d66a20640c328d1c336d9115ca1787703ea1f48)

That test is no longer necessary because of the stricter condition asserted later.